CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+# Contributing to this repository
+
+# Table of Contents
+1. [Commits and pull requests](#commits-release-notes-pull-requests)
+    1. [Fixing a bug](#fixing-a-bug)
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests
+### Fixing a bug
+
+1. Fork the repo and commit changes to a new branch in your fork.
+2. Make pull requests against the master branch in the rpc-gating repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure:
+
+* The first line should be limited to 50 characters, should not end with a full stop and the first word should be capitalised.
```

Comment:
This is an arbitrary limit which seems to be derived from https://chris.beams.io/posts/git-commit/ but applied in a draconian way. 50 chars is sometimes not enough. I'd say ideally no more than 50, but definitely no more than 72 (git will wrap it anyway).

Reply:
@odyssey4me yeah the original rpco commit message guidelines were _inspired_ by that post. I'm not sure about the Draconian nature of wanting good grammar in the subject line, but what do you want the text to say regarding the character limit?

Reply:
I suggest referencing that post, as it's a good read for background. But then we should rather say something like ``The first line should ideally be limited to 50 characters, but must be no more than 70 characters. It should not end with a full stop and the first word should be capitalised.``